File: backend_node/src/models/Address.js
const { DataTypes } = require('sequelize');
const sequelize = require('../config/db');

/**
 * Address Model
 * Represents a saved address for quick checkout reuse
 */
const Address = sequelize.define('Address', {
  id: {
    type: DataTypes.UUID,
    defaultValue: DataTypes.UUIDV4,
    primaryKey: true,
  },
  user_id: {
    type: DataTypes.UUID,
    allowNull: false,
    comment: 'Customer ID',
  },
  label: {
    type: DataTypes.STRING(50),
    allowNull: false,
    comment: 'e.g. Home, Work, Other',
  },
  address: {
    type: DataTypes.TEXT,
    allowNull: false,
  },
  city: {
    type: DataTypes.STRING(100),
    allowNull: false,
  },
  building: {
    type: DataTypes.STRING(100),
    allowNull: true,
  },
  floor: {
    type: DataTypes.STRING(50),
    allowNull: true,
  },
  notes: {
    type: DataTypes.TEXT,
    allowNull: true,
  },
  created_at: {
    type: DataTypes.DATE,
    defaultValue: DataTypes.NOW,
  },
}, {
  tableName: 'addresses',
  timestamps: false,
});

module.exports = Address;

File: frontend_flutter/lib/models/address.dart
// =============================================================================
// NADDEFLI — address.dart
// Layer: Flutter — Model
// Purpose: Data class for a saved customer address.
// Connects to: Parsed from /api/addresses
// =============================================================================

/// Saved Address Model
class Address {
  final String id;
  final String userId;
  final String label; // Home, Work, Other
  final String address;
  final String city;
  final String? building;
  final String? floor;
  final String? notes;
  final DateTime? createdAt;

  Address({
    required this.id,
    required this.userId,
    required this.label,
    required this.address,
    required this.city,
    this.building,
    this.floor,
    this.notes,
    this.createdAt,
  });

  /// Factory constructor for JSON deserialization
  factory Address.fromJson(Map<String, dynamic> json) {
    return Address(
      id: json['id'] ?? '',
      userId: json['user_id'] ?? '',
      label: json['label'] ?? 'Home',
      address: json['address'] ?? '',
      city: json['city'] ?? '',
      building: json['building'],
      floor: json['floor'],
      notes: json['notes'],
      createdAt: json['created_at'] != null
          ? DateTime.tryParse(json['created_at'].toString())
          : null,
    );
  }

  /// Convert to JSON
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'user_id': userId,
      'label': label,
      'address': address,
      'city': city,
      'building': building,
      'floor': floor,
      'notes': notes,
      'created_at': createdAt?.toIso8601String(),
    };
  }
}
